File: js/charCount.js
exports.charCount = function(str) {

    const getCount=(ltr, str)=>{
        let count=0
        for(let i=0; i<str.length; i++){
            if(str[i]==ltr){
                count++
            }
        }
        return count;
    }

    let manStr=str.replaceAll(" ","")
    manStr=str.split('')
    let listOfLtrs=[]
    let listOfInst=[]
    let answer=[]
    for(let i=0; i< manStr.length; i++){
        if (listOfLtrs.includes(manStr[i])){
            continue
        }
        else{
            listOfLtrs.push(manStr[i])
            listOfInst.push(getCount(manStr[i], manStr))
        }
    }
    listOfInst=listOfInst.sort().reverse()
    listOfLtrs=listOfLtrs.sort()
    for (let j=0; j<listOfInst.length; j++){
        for(let k=0; k<listOfLtrs.length; k++){
            if (getCount(listOfLtrs[k], manStr)==listOfInst[j] && listOfLtrs[k] !==" "){
                answer.push([listOfLtrs[k],listOfInst[j]])
                listOfLtrs[k]=" "
                listOfInst[j]=0
            }
        }
    }
    return answer;
};

File: python/char_count.py
def char_count(stri):
    man_str=stri.replace(" ","")
    list_of_chars=[]
    list_of_inst=[]
    answer=[]
    man_str=[*man_str]
    for ltr in man_str:
        if ltr in list_of_chars:
            pass
        else:
            list_of_chars.append(ltr)
            list_of_inst.append(man_str.count(ltr))
    list_of_chars=sorted(list_of_chars)
    list_of_inst=sorted(list_of_inst, reverse=True)
    for inst in range(len(list_of_inst)):
        for ltr in range(len(list_of_chars)):
            if man_str.count(list_of_chars[ltr])==list_of_inst[inst] and list_of_chars[ltr] !=" ":
                answer.append([list_of_chars[ltr], list_of_inst[inst]])
                list_of_chars[ltr] =" "
                list_of_inst[inst] = 0
    return answer